### PR TITLE
Ung 2343 get poc admin status

### PR DESCRIPTION
**Pull request by wojciechUrbanski**

### CODE REVIEW

File: src/main/scala/com/ubirch/controllers/TenantAdminController.scala [hunk 1]
```diff
@@ -135,178 +135,159 @@ class TenantAdminController @Inject() (
...
+            NotFound(NOK.resourceNotFoundError("Could not find PoC Admin status with provided ID"))
+          case Left(PocAdminStatusNotFound(pocAdminId)) =>
+            logger.error(s"Could not find PoC Admin Status for PoC Admin with id $pocAdminId")
+            NotFound(NOK.resourceNotFoundError("Could not find Poc Admin Status assigned to provided PoC Admin"))
+        }
+      }
+    }
+  }
+
+  private def tenantAdminEndpoint(description: String)(logic: Tenant => Task[ActionResult]) = {
```
Comment (gzhk) [inline src/main/scala/com/ubirch/controllers/TenantAdminController.scala:285]: 👍🏼

File: src/main/scala/com/ubirch/controllers/TenantAdminController.scala [hunk 2]
```diff
@@ -315,6 +296,17 @@ class TenantAdminController @Inject() (
     }
   }
 
+  private def getParamAsUUID(paramName: String, errorMsg: String => String)(logic: UUID => Task[ActionResult]) = {
+    val id = params(paramName)
+    Try(UUID.fromString(id)) match {
+      case Success(uuid) => logic(uuid)
+      case Failure(ex) =>
+        logger.error(errorMsg(id), ex)
+        Task(InternalServerError(NOK.serverError(errorMsg + ex.getMessage)))
```
Comment (gzhk) [inline src/main/scala/com/ubirch/controllers/TenantAdminController.scala:305]: IMO it should be a bad request

Comment (wojciechUrbanski) [reply src/main/scala/com/ubirch/controllers/TenantAdminController.scala]: Yup :) Changed to BadRequest

File: src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala [hunk 3]
```diff
@@ -119,14 +129,32 @@ class DefaultTenantAdminService @Inject() (
...
     EitherT(pocAdminRepository.getPocAdmin(pocAdminId).map(_.toRight(error(pocAdminId))))
   }
+  override def getPocAdminStatus(
+    tenant: Tenant,
+    pocAdminId: UUID): Task[Either[GetPocAdminStatusErrors, GetPocAdminStatusResponse]] = {
+    import GetPocAdminStatusErrors._
+    (for {
+      pocAdmin <- getPocAdmin(pocAdminId, adminId => PocAdminNotFound(adminId))
+      _ <-
+        isPocAdminAssignedToTenant[Task, GetPocAdminStatusErrors](tenant, pocAdmin)(
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/services/tenantadmin/TenantAdminService.scala:139]: we should add a similar logic to the GET /pocStatus. Thanks for being as accurate!

**Review by gzhk — APPROVED** [review]

**Review by lalottacontinua — APPROVED** [review]

**Pull request merged**